packages/core/rpc/RpcManager.ts

```diff
@@ -49,34 +58,72 @@ export default class RpcManager {
     this.driverObjects = new Map()
   }
 
-  getDriver(backendName: keyof typeof DriverClasses): DriverClass {
-    const driver = this.driverObjects.get(backendName)
+  getDriver(driverName: string): DriverClass {
+    const driverConfig = this.mainConfiguration.drivers.get(driverName)
+    if (!driverConfig) {
+      throw new Error(`could not find driver config for "${driverName}"`)
+    }
+    const driver = this.driverObjects.get(driverName)
     if (driver) return driver
 
-    const backendConfiguration = this.backendConfigurations[backendName]
-    const DriverClassImpl = DriverClasses[backendName]
-
-    if (!DriverClassImpl) {
+    // a bit verbose, but it keeps TypeScript happy
+    let newDriver
+    const backendName = driverConfig.type as keyof typeof DriverClasses
+    if (backendName === 'MainThreadRpcDriver') {
+      const backendConfiguration = this.backendConfigurations
+        .MainThreadRpcDriver
+      if (!backendConfiguration) {
+        throw new Error(
+          `requested RPC driver "${backendName}" is missing config`,
+        )
+      }
+      const DriverClass = DriverClasses[backendName]
+      newDriver = new DriverClass({
+        ...backendConfiguration,
+        config: driverConfig,
+      })
+    } else if (backendName === 'ElectronRpcDriver') {
+      const backendConfiguration = this.backendConfigurations.ElectronRpcDriver
+      if (!backendConfiguration) {
+        throw new Error(
+          `requested RPC driver "${backendName}" is missing config`,
+        )
+      }
+      const DriverClass = DriverClasses[backendName]
+      newDriver = new DriverClass(
+        { ...backendConfiguration, config: driverConfig },
+        { plugins: this.runtimePluginDefinitions },
+      )
+    } else if (backendName === 'WebWorkerRpcDriver') {
+      const backendConfiguration = this.backendConfigurations.WebWorkerRpcDriver
+      if (!backendConfiguration) {
+        throw new Error(
+          `requested RPC driver "${backendName}" is missing config`,
+        )
+      }
+      const DriverClass = DriverClasses[backendName]
+      newDriver = new DriverClass(
+        { ...backendConfiguration, config: driverConfig },
+        { plugins: this.runtimePluginDefinitions },
+      )
+    } else {
       throw new Error(`requested RPC driver "${backendName}" is not installed`)
-    } else if (!backendConfiguration) {
-      throw new Error(`requested RPC driver "${backendName}" is missing config`)
     }
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const newDriver = new DriverClassImpl(backendConfiguration as any, {
-      plugins: this.runtimePluginDefinitions,
-    })
-    this.driverObjects.set(backendName, newDriver)
+
+    this.driverObjects.set(driverName, newDriver)
     return newDriver
   }
```

Comment:
i experimented just now with reverting getDriver to the implementation in master and typescript seemed OK with it, could you check that this is really necessary

Reply:
I remember I was trying to get rid of a TypeScript error with this, but now I can't reproduce it. I've reverted it.